src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordCopyCtor.cs

```diff
@@ -45,11 +48,26 @@ internal override void GenerateMethodBodyStatements(SyntheticBoundNodeFactory F,
             var param = F.Parameter(Parameters[0]);
             foreach (var field in ContainingType.GetFieldsToEmit())
             {
-                if (!field.IsStatic)
+                if (!field.IsStatic && field.ContainingType.Equals(ContainingType))
                 {
                     statements.Add(F.Assignment(F.Field(F.This(), field), F.Field(param, field)));
                 }
             }
         }
+
+        internal static MethodSymbol? FindCopyConstructor(NamedTypeSymbol containingType)
+        {
+            foreach (var member in containingType.InstanceConstructors)
+            {
+                if (member is MethodSymbol { IsStatic: false, ParameterCount: 1, Arity: 0 } method &&
+                    method.Parameters[0].Type.Equals(containingType, TypeCompareKind.CLRSignatureCompareOptions) &&
+                    method.Parameters[0].RefKind == RefKind.None)
+                {
+                    return method;
```

Comment:
>return method; [](start = 20, length = 14)

I think we should ignore constructors with unsupported metadata.

Reply:
Can you clarify? Do you mean a modreq for instance?

---
In reply to: [438862580](https://github.com/dotnet/roslyn/pull/45022#discussion_r438862580) [](ancestors = 438862580)

Reply:
> Can you clarify? Do you mean a modreq for instance?

Yes. Check out HasUnsupportedMetadata property

---
In reply to: [438935311](https://github.com/dotnet/roslyn/pull/45022#discussion_r438935311) [](ancestors = 438935311,438862580)